src/algorithms/generate.rs

```diff
@@ -93,14 +95,14 @@ pub(crate) fn generate_multi_prime_key_with_exp<R: CryptoRngCore + ?Sized>(
 
         let n = compute_modulus(&primes);
 
-        if n.bits() != bit_size {
+        if n.bits() as usize != bit_size {
             // This should never happen for nprimes == 2 because
             // gen_prime should set the top two bits in each prime.
```

Comment:
The comment should be updated for the new name of the function.

Also I don't quite understand "gen_prime should set the top two bits in each prime" - what was that behavior for? `generate_prime_with_rng()` only sets one top bit, to ensure that the returned primes are of the requested bit size.

Reply:
I was a bit curious about that myself. I saw @FiloSottile mention the same thing in this post:

https://words.filippo.io/dispatches/rsa-keygen-bench/

> There is almost nothing special to selecting prime candidates. You draw an appropriately sized random number from a CSPRNG, and to avoid wasting time, you set the least significant bit and the two most significant bits: large even numbers are not prime, and setting the top two guarantees N won’t come out too small.

Reply:
If you set only the top bit of a n/2 bits number, you get a value >= 2^(n/2-1). If you multiply two of them, you get a value >= 2^(n-2), which is at a minimum n-1 bits long, one too short for a n-bit modulus.

Reply:
Ah I see, that makes sense. Do you think it should be the default behavior in `crypto-primes`?

Reply:
If you mainly need the primes for RSA, yes, you are only wasting CPU rejecting small moduli after you found the primes. If you're using the primes for something else, it might depend on what they are for.

Reply:
@fjarri are you planning on changing the behaviour?

Reply:
I thought about it, and I don't think it should be the default - too specialized. It can be achieved with custom sieves, but I need to make a release first

Reply:
that's fair, please do let me know if you have a release/how to make replicate the desired behavior with custom sieves

Reply:
Opened https://github.com/entropyxyz/crypto-primes/pull/69. If that is merged, you will need to use `sieve_and_find()` to access this functionality.

Reply:
Nice! Thanks for `SetBits::TwoMsb`

Reply:
done, thank you